src/selenium/schedulePage.js

```diff
@@ -0,0 +1,139 @@
+    return self.find(By.className('calendar-view')).findElements(By.className('calendar')).then(self.getElemsDisplayStatus);
+  }
+
+  return null;
+};
+
+SchedulePage.getCurrentView = function() {
+  // Gives the current status about visibility of the date divs inside the list and the calendar container
+  const promiseArr = [];
+
+  promiseArr.push(this.getVisibleDates('list'));
+  promiseArr.push(this.getVisibleDates('calendar'));
+  return Promise.all(promiseArr);
+};
+
+SchedulePage.changeDay = function(dayNo) {
+  const self = this;
+  const dateId = datesId[dayNo - 1];
+
+  return self.find(By.id(dateId)).then(self.click).then(self.getCurrentView.bind(self));
+};
+
+SchedulePage.toggleMode = function() {
+  const self = this;
+  const toggleButtonId = 'page-mode';
+
+  return self.find(By.id(toggleButtonId)).then(self.click).then(self.getCurrentView.bind(self));
+};
+
+SchedulePage.getDownloadDropdown = function() {
+  const self = this;
+  const promiseArr = [];
+
+  const clickDropdownButton = function() {
+    return self.find(By.css('.filter.dropdown .fa-download')).then(function(icon) {
+      return icon.findElement(By.xpath('..'));
+    }).then(function(button) {
+      return button.click().then(self.driver.sleep(1000));
+    });
+  };
+
+  return Promise.resolve().then(function() {
+    return clickDropdownButton();
+  }).then(function() {
+    promiseArr.push(self.find(By.className('download-dropdown')).isDisplayed());
+  }).then(function() {
+    return clickDropdownButton();
+  }).then(function() {
+    promiseArr.push(self.find(By.className('download-dropdown')).isDisplayed());
+  }).then(function() {
+    return Promise.all(promiseArr);
+  });
+};
+
+SchedulePage.checkFilterDirectLink = function() {
+  const self = this;
+  const roomTrackIdArr = [];
+
+  function pushId(roomTrackElem) {
+    roomTrackElem.getAttribute('id').then(function(id) {
+      roomTrackIdArr.push(id);
+    });
+  }
+
+  return new Promise(function(resolve) {
+    self.findAll(By.className('schedule-track')).then(function(roomTrackElems) {
+      roomTrackElems.forEach(function(roomTrackElem) {
+        roomTrackElem.isDisplayed().then(function(val) {
+          if (val === true) {
+            pushId(roomTrackElem);
+          }
+        });
+      });
+    }).then(function() {
+      resolve(roomTrackIdArr);
+    });
+  });
+};
+
+SchedulePage.checkFilterDynamicLink = function() {
+  const self = this;
+  const promiseArr = [];
+
+  promiseArr.push(self.activeRooms());
+  promiseArr.push(self.activeTracks());
+  return Promise.all(promiseArr);
+};
+
+SchedulePage.checkVideo = function() {
+  const self = this;
+
+  return new Promise(function(resolve) {
+    self.find(By.id('title-3015')).then(self.click).then(function() {
+      self.find(By.id('video-3015')).then(function(elem) {
+        elem.getAttribute('src').then(function(url) {
+          resolve(self.isLinkBroken(url));
+        });
+      });
+    });
+  });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Block must be padded by blank lines. (padded-blocks)](https://app.codacy.com/manual/triplez/open-event-webapp/pullRequest?prid=3226522)